custom_completions.go

```diff
@@ -150,6 +182,33 @@ func (c *Command) getCompletions(args []string) (*Command, []string, ShellCompDi
 		}
 	}
 
+	if flag == nil {
+		// Complete subcommand names
```

Comment:
I was writing this when you merged:

I must point out that the logic in this block completes like this: 
- (`sub-commands` AND `ValidArgs`) XOR (`sub-commands` AND `ValidArgsFunction`) 

which is not how the bash completion currently does it.
It is instead based on #1088.  If you do not agree, I can make the necessary changes, but I do feel the current for bash behaviour is not correct.

Reply:
Let's let it ride for right now and see if any issues get opened for it.